### PR TITLE
Reactor improvements

### PR DESCRIPTION
**Pull request by alshopov**
Start of improvements on build system

I intend to:
eliminate maven-antrun-plugin in preference to maven thrift plugin.
change code to enable updates of dependencies

### CODE REVIEW

**Comment by alshopov** [comment]
Hmm, will have to look at the build failure

File: tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java [hunk 1]
```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
...
+        for (Connection conn : connections.values()) {
+            if (conn.direction == Connection.Direction.OUT) {
                 out++;
             } else {
                 in++;
             }
         }
 
-        Map<String, Integer> result = new HashMap<>();
+        Map<String, Integer> result = new HashMap<>(2);
```
Comment (yborovikov) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: Unfortunately, passing `2` to the constructor won't allow the map to hold 2 entries without re-sizing (have to account for the load factor, which is `0.75` by default). Has to be something like:
```
public static int hashMapCapacity(int size) { return (size * 4 + 2) / 3; } // for load factor of 0.75
...
   new HashMap(hashMapCapacity(2), 0.75f);
```

Comment (alshopov) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: You are right about the 2 and resize but the reason is different: while 2 is enough, there is an additional internal HashMap call to resize when the capacity has been reached. I am making it 3. The load factor is not a problem.

File: tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java [hunk 2]
```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
     }
 
     public void close() {
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn != null) {
-                conn.close();
-            }
+        for ( Connection conn : connections.values()) {
```
Comment (manasdk) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: nit : space betweeen `( Connection`

File: tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java [hunk 3]
```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
...
+            conn.close();
```
Comment (manasdk) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: null check not required any longer?

File: tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java [hunk 4]
```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
...
     public Map<String, Integer> getStats() {
         int in = 0;
         int out = 0;
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn == null) {
-                continue;
-            } else if (conn.direction == Connection.Direction.OUT) {
+        for (Connection conn : connections.values()) {
+            if (conn.direction == Connection.Direction.OUT) {
```
Comment (manasdk) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: was removal of null check intentional?

File: tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java (hunk 3 above)
Comment (alshopov) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: Yes. connections is ConcurrentHashMap. It cannot be null. 
Like Hashtable but unlike HashMap, this class does not allow null to be used as a key or value.
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html

File: tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java (hunk 4 above)
Comment (alshopov) [inline tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java]: Yes. connections is ConcurrentHashMap. It cannot be null. 
Like Hashtable but unlike HashMap, this class does not allow null to be used as a key or value.
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html

**Pull request merged**